src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -2945,16 +2945,22 @@ checkNullable:
                 Dim identifierNameOpt As IdentifierNameSyntax = Nothing
                 Dim asKeywordOpt As KeywordSyntax = Nothing
 
+                ' if there is a type character or As, then this must be a name
                 If CurrentToken.Kind = SyntaxKind.IdentifierToken AndAlso
```

Comment:
Added some notes to [https://github.com/dotnet/roslyn/issues/14293](https://github.com/dotnet/roslyn/issues/14293)


---
In reply to: [82065594](https://github.com/dotnet/roslyn/pull/14284#discussion_r82065594) [](ancestors = 82065594,82047853)